src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/CharMarshallingGeneratorFactory.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Interop
+{
+    public sealed class CharMarshallingGeneratorFactory : IMarshallingGeneratorFactory
+    {
+        private static readonly IMarshallingGenerator s_blittable = new BlittableMarshaller();
+        private static readonly IMarshallingGenerator s_utf16Char = new Utf16CharMarshaller();
+
+        private readonly IMarshallingGeneratorFactory _inner;
+        private readonly bool _useBlittableMarshallerForUtf16;
+
+        public CharMarshallingGeneratorFactory(IMarshallingGeneratorFactory inner, bool useBlittableMarshallerForUtf16)
+        {
+            _inner = inner;
+            _useBlittableMarshallerForUtf16 = useBlittableMarshallerForUtf16;
+        }
+
+        public IMarshallingGenerator Create(TypePositionInfo info, StubCodeContext context)
+        {
+            if (info.ManagedType is SpecialTypeInfo { SpecialType: SpecialType.System_Char })
+            {
+                return CreateCharMarshaller(info, context);
+            }
+
+            return _inner.Create(info, context);
+        }
+
+        private IMarshallingGenerator CreateCharMarshaller(TypePositionInfo info, StubCodeContext context)
+        {
+            MarshallingInfo marshalInfo = info.MarshallingAttributeInfo;
+            if (marshalInfo is NoMarshallingInfo)
+            {
+                // [Compat] Require explicit marshalling information.
+                throw new MarshallingNotSupportedException(info, context)
+                {
+                    NotSupportedDetails = SR.MarshallingStringOrCharAsUndefinedNotSupported
+                };
+            }
+
+            // Explicit MarshalAs takes precedence over string encoding info
+            if (marshalInfo is MarshalAsInfo marshalAsInfo)
+            {
+                switch (marshalAsInfo.UnmanagedType)
+                {
+                    case UnmanagedType.I2:
+                    case UnmanagedType.U2:
+                        // If runtime marshalling is disabled, we treat UTF-16 char as blittable
```

Comment:
Remove the comment? Not that you have separated runtime marshalling being disabled from this setting. Same below.